File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/TimeAccumulator.java
package Auxiliary_Modules;

public class TimeAccumulator {
    // accumulator ++ each tick minutes interval
    private TimeGate timeGate;
    private int accumulator = 0;
    public void setTick(int tick) {
        timeGate.setPause(tick);
    }

    public TimeAccumulator(int tick) {
        // accumulation ticker
        this.timeGate = new TimeGate(tick);
        timeGate.openGate();
    }

    public int getAccumulator() {
        return accumulator;
    }
    public void tick(){
        if (timeGate.isClosed()){
            timeGate.openGate();
            accumulator ++;
        }
    }
    public void reset(){accumulator = 0;}

    public void decAccumulator() {
        if (accumulator > 0){accumulator--;}
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/TimeAccumulator.kt
package auxiliary_modules

class TimeAccumulator(tick: Int) {
    // accumulator ++ each tick minutes interval
    private var timeGate = TimeGate(5)
    var accumulator = 0
        private set

    fun setTick(tick: Int) {
        timeGate.setPause(tick)
    }

    init {
        // accumulation ticker
        timeGate = TimeGate(tick)
        timeGate.openGate()
    }

    fun tick() {
        if (timeGate.isClosed) {
            timeGate.openGate()
            accumulator++
        }
    }

    fun reset() {
        accumulator = 0
    }

    fun decAccumulator() {
        if (accumulator > 0) {
            accumulator--
        }
    }
}